ocs_ci/ocs/longevity_stage2.py

```diff
@@ -0,0 +1,563 @@
+):
+    """
+    Write empty files to bucket and verify if they are created.
+
+    Args:
+        mcg_obj (MCG) : An MCG object containing the MCG S3 connection credentials
+        awscli_pod_session : Fixture to create a new AWSCLI pod for relaying commands.
+        bucket_name (str) : Name of the bucket on which files are to be written.
+        test_directory_setup : Fixture to setup test DIRs.
+
+    Raises:
+        UnexpectedBehaviour : Raises an exception if files are not created.
+
+    Returns:
+        Set: A set of names of all bucket objects.
+
+    """
+
+    full_object_path = f"s3://{bucket_name}"
+    data_dir = test_directory_setup.origin_dir
+
+    # Touch create 1000 empty files in bucket
+    command = f"for file_no in $(seq 1 1000); do touch {data_dir}/test$file_no; done"
+    awscli_pod_session.exec_sh_cmd_on_pod(command=command, sh="sh")
+    # Write all empty objects to the bucket
+    sync_object_directory(awscli_pod_session, data_dir, full_object_path, mcg_obj)
+
+    logger.info("Successfully created files.")
+
+    obj_set = set(obj.key for obj in mcg_obj.s3_list_all_objects_in_bucket(bucket_name))
+    test_set = set("test" + str(file_no + 1) for file_no in range(1000))
+
+    if test_set != obj_set:
+        raise ex.UnexpectedBehaviour("File name set does not match")
+    logger.info("File name set match")
+
+    return obj_set
+
+
+def measure_attach_time(pod_objs, pvc_size):
+    """
+    Measures and Logs Attach Time of all PODs.
+
+    Args:
+        pod_objs (list) : List of POD objects for which we have to measure the time.
+        pvc_size (int) : Size of each PVC in GB.
+
+    Raises:
+        PerformanceException : Raises an exception if POD attach time is greater than the accepted time.
+
+    Logs:
+        Attach time of all PODs, as well as the average time.
+
+    """
+    msg_prefix = f"PVC size: {pvc_size}."
+    start_time_dict_list = []
+    for pod in pod_objs:
+        start_time_dict_list.append(helpers.pod_start_time(pod))
+    logger.info(str(start_time_dict_list))
+    time_measures = []
+    for attach_time in start_time_dict_list:
+        if "my-container" in attach_time:
+            time_measures.append(attach_time["my-container"])
+        else:
+            time_measures.append(attach_time["web-server"])
+    for index, start_time in enumerate(time_measures):
+        logger.info(
+            f"{msg_prefix} pod number {index + 1} start time: {start_time} seconds"
+        )
+        if start_time > 30:
+            raise ex.PerformanceException(
+                f"{msg_prefix} Pod number {index} start time is {start_time},"
+                f"which is greater than 30 seconds"
+            )
+    if time_measures:
+        average = statistics.mean(time_measures)
+        logger.info(
+            f"{msg_prefix} The average time for the sampled {len(time_measures)} pods is {average} seconds."
+        )
+
+
+def measure_pod_creation_time(namespace, num_of_pods):
```

Comment:
I feel it would be better to check the pod creation time for each pod_name than with num_of_pods, wdyt @PrasadDesala @malhotra700 ?

Reply:
Yes, it would be better to check the creation time for each pod, but the time complexity would increase as we would have to iterate through the logs for each pod. Currently, I am iterating through the logs only once.

Reply:
+1 to Akarsha.
It will be more helpful if the function can measure the pod creation time for a specific pod or list of pods.
Measuring pod creation time is less priority atleast from the Longevity perspective, so this can be taken in the separate PR